detector.go

```diff
@@ -301,10 +308,14 @@ func (bo BuildpackOrder) Detect(c *DetectConfig) (BuildpackGroup, BuildPlan, err
 
 func (bo BuildpackOrder) detect(done, next []Buildpack, optional bool, wg *sync.WaitGroup, c *DetectConfig) ([]Buildpack, []BuildPlanEntry, error) {
 	ngroup := BuildpackGroup{Group: next}
+	hadErr := false
 	for _, group := range bo {
 		// FIXME: double-check slice safety here
 		found, plan, err := group.append(ngroup).detect(done, wg, c)
-		if err == ErrFail {
+		if err == ErrFailWithErrors {
```

Comment:
I did not like having `ErrFailWithErrors` in so many places. I wonder if making a single `ErrFail ` with a flag for `DetectErr` or something would be cleaner.

Reply:
I don't have strong opinions on this one.

Someday I could see us having an umbrella `lifecycle.Error` type with a field to denote the subspecies of error. But maybe it's unnecessary when there are only two?

Reply:
Sounds fine to me! I like the idea of the umbrella error type in the future.